datamol/reactions/_reactions.py

```diff
@@ -0,0 +1,290 @@
+
+    return rdChemReactions.SanitizeRxn(rxn) in [
+        rdChemReactions.SanitizeFlags.SANITIZE_NONE,
+        None,
+    ]
+
+
+def select_reaction_output(
+    product: list,
+    single_output: bool = True,
+    rm_attach: bool = False,
+    as_smiles: bool = False,
+    sanitize: bool = True,
+) -> Union[list, str, dm.Mol]:
+    """
+    Compute the products from a reaction. It only takes the first product of the
+
+    Args:
+        product: All the products from a reaction.
+        single_output: Whether return a single output from a reaction.
+        rm_attach: Whether remove the attachment point from the product.
+        as_smiles: Whether return the result in smiles.
+        sanitize: Whether sanitize the product to return.
+
+    Returns:
+        Processed products from reaction.
+    """
+    # flatten all possible products of a reaction
+    product = list(sum(product, ()))
+    if single_output:
+        product = list(np.random.choice(product, 1))
+    if sanitize:
+        product = [dm.sanitize_mol(m) for m in product]
+    if rm_attach:
+        product = [dm.remove_dummies(x) for x in product]
+    if as_smiles:
+        product = [dm.to_smiles(x) for x in product if x is not None]
+    if single_output:
+        return product[0]
+    return product
+
+
+def apply_reaction(
+    rxn: dm.ChemicalReaction,
+    reactants: tuple,
+    single_output: bool = False,
+    as_smiles: bool = False,
+    rm_attach: bool = False,
+    disable_logs: bool = True,
+    sanitize: bool = True,
+) -> Union[list, str, dm.Mol]:
+    """
+    Apply a chemical reaction on a molecule
+
+    Args:
+       rxn: Reaction object.
+       reactants: A tuple of reactants.
+       single_output: Whether return one product from all possible product.
+       as_smiles: Whether return product in SMILES.
+       rm_attach: Whether remove the attachment point from product.
+       disable_logs: Whether disable rdkit logs.
+       sanitize: Whether sanitize the product.
+
+    Returns:
+       Reaction products.
+    """
+    with dm.without_rdkit_log(enable=disable_logs):
+        if not rxn.IsInitialized():
+            rxn.Initialize()  # pragma: no cover
+
+        product = rxn.RunReactants(reactants)
+        outputs = select_reaction_output(
+            product=product,
+            single_output=single_output,
+            as_smiles=as_smiles,
+            rm_attach=rm_attach,
+            sanitize=sanitize,
+        )
+
+    return outputs
+
+
+def can_react(rxn: dm.ChemicalReaction, mol: dm.Mol) -> bool:
+    """
+    Check if a molecule is a reactant to a chemical reaction and return position
+
+    Args:
+        rxn: Reaction to check.
+        mol: Molecule to check if it is a reactant.
+
+    Returns:
+        Position of mol
+    """
+    if not rxn.IsInitialized():
+        rxn.Initialize()  # pragma: no cover
+    if rxn.IsMoleculeReactant(mol):
+        return find_reactant_position(rxn, mol) != -1
+    return False
```

Comment:
It now returns a boolean and not a mix of bool and int. To get the position you can use `find_reactant_position`.